module/VuFind/src/VuFind/Auth/OpenIDConnect.php

```diff
@@ -0,0 +1,520 @@
+<?php
+
+/**
+ * OpenID Connect authentication module.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) R-Bit Technology 2018-2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Authentication
+ * @author   Josef Moravec <josef.moravec@gmail.com>
+ * @author   Radek Šiman <rbit@rbit.cz>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFind\Auth;
+
+use DomainException;
+use Firebase\JWT\BeforeValidException;
+use Firebase\JWT\ExpiredException;
+use Firebase\JWT\JWK;
+use Firebase\JWT\JWT;
+use Firebase\JWT\SignatureInvalidException;
+use InvalidArgumentException;
+use Laminas\Session\Container as SessionContainer;
+use UnexpectedValueException;
+use VuFind\Db\Row\User;
```

Comment:
To make this more future-proof, I'd suggest using VuFind\Db\Entity\UserEntityInterface instead of VuFind\Db\Row\User so changes will not be needed when #2233 is completed.

Reply:
Fixed